### ISSUE TITLE
SDist PKG-INFO file should include Requires-Dist entries.

### ISSUE THREAD

**Issue by ales-erjavec** [pypa/setuptools#1716]
setuptools already patches base distutils [`write_pkg_file`](https://github.com/pypa/setuptools/blob/16e452a/setuptools/dist.py#L126)) file to include extra meta data for core metadata specification 1.2 and 2.1 (adding `Requires-Python`, `Description-Content-Type`, `Project-Urls`, `Provides-Extra` entries) .

It would be nice if it could also generate the [`Requires-Dist`](https://packaging.python.org/specifications/core-metadata/#requires-dist-multiple-use) entries (along with extras and env markers).

**Comment by pganssle** [comment]
Yes, I generally agree that we should be implementing the full 2.1 core metadata specification. It seems to me that when you build a wheel, the `Requires-Dist` lines *are* populated, I'm guessing that `wheel` has its own implementation or something?

I'm guessing we should do basically the same thing `wheel` does. A PR fixing this may be pretty test-heavy.

**Comment by bhrutledge** [comment]
Is this a duplicate of https://github.com/pypa/setuptools/issues/1805?
